frontend/src/components/core/StreamlitDialog/ThemeCreator.tsx

```diff
@@ -21,43 +20,56 @@ export interface Props {
   hasCustomTheme: boolean
 }
 
-interface ThemeBuilder {
-  [key: string]: {
-    desc: string
-    title: string
-    component: any
-    options?: any[]
-  }
+interface ThemeOptionBuilder {
+  desc: string
+  title: string
+  component: any
+  options?: any[]
+  getValue: (value: string, config: ThemeOptionBuilder) => any
 }
 
-const themeBuilder: ThemeBuilder = {
+const valueToColor = (value: string, config: ThemeOptionBuilder): string =>
```

Comment:
nit: would be nice to mark `config` as intentionally unused by writing `_config` (assuming typescript doesn't care about the actual names of parameters in types)